### PR TITLE
Support: FCMPLayer; WeightTable; ONNX transpose convo.

### PR DESCRIPTION
**Pull request by wenyushi**
The push involves changes as below:

- Support FCMPLayer definition.
- Support WeightsTable which provides a flexible way to generate a new weights table for a model given an existing weights table.
- Support importing ONNX group convolution; add norm_std option for the input layer; parse group option for convolution operation.
- Fix several issues regarding import weights from external files and check file dependency in test cases.
- Fix a bug related to caslib.

### CODE REVIEW

File: dlpy/layers.py [hunk 1]
```diff
@@ -1680,7 +1689,7 @@ class OutputLayer(Layer):
         Default: AUTO
     fcmp_act : string, optional
         Specifies the FCMP activation function for the layer.
-    fcmp_error : string, optional
```
[the remark this answers is not in the document]
Comment (wenyushi) [reply dlpy/layers.py]: The change is not for making it short. fcmp_error cannot be correctly parsed. On CAS side, fcmpErr is the one we have. So, it is a bug to be fixed.

[the remark this answers is not in the document]
Comment (ethem-kinginthenorth) [reply dlpy/layers.py]: OK.

File: dlpy/layers.py [hunk 2]
```diff
@@ -1466,7 +1466,12 @@ def num_bias(self):
         if self._num_bias is None:
             if self.src_layers is None:
                 self._num_bias = 0
-            self._num_bias = int(2 * self.src_layers[0].output_size[2])
+            # if the input of layer is in hwc
```
Comment (ethem-kinginthenorth) [inline dlpy/layers.py]: please explain h x w x c, e.g., 3 dimensional (in the h x w x c format).

File: dlpy/layers.py [hunk 3]
```diff
@@ -1466,7 +1466,12 @@ def num_bias(self):
...
+            if isinstance(self.src_layers[0].output_size, Iterable):
+                self._num_bias = int(2 * self.src_layers[0].output_size[-1])
+            # if the input of layer is one dimension
```
Comment (ethem-kinginthenorth) [inline dlpy/layers.py]: dimensional

File: dlpy/network.py [hunk 4]
```diff
@@ -748,16 +754,27 @@ def set_weights(self, weight_tbl):
...
 
         '''
-        weight_tbl = input_table_check(weight_tbl)
         weight_name = self.model_name + '_weights'
 
-        if weight_tbl['name'].lower() != weight_name.lower():
-            self._retrieve_('table.partition',
-                            casout=dict(replace=True, name=self.model_name + '_weights'),
-                            table=weight_tbl)
+        if type(weight_tbl) == WeightsTable:
```
[the remark this answers is not in the document]
Comment (ethem-kinginthenorth) [reply dlpy/network.py]: can you please also explain why we need remapping. thanks

File: dlpy/network.py [hunk 5]
```diff
@@ -1577,6 +1597,133 @@ def count_params(self):
...
+    A weights table builds connection with a deep learning model.
+    One use case is a new model setting a pre-trained weights. Since SAS deep learning model loads weights according to
+    layer id and the order of layer id might be different between the new built model and the pre-trained model.
+    So, the instance of the class can remap layer id and generate a suitable weights table according to layer names.
+    Example:
+        pretrained_weights_table = WeightsTable(conn, weights_tbl_name='my_pretrained_weights_table',
+                                                model_tbl_name='my_pretrained_model_table')
+        new_model.set_weights(pretrained_weights_table)
+
+    Parameters
```
Comment (ethem-kinginthenorth) [inline dlpy/network.py]: this is good. thanks

**Pull request merged**